src/main/java/org/apache/spark/util/collection/unsafe/sort/PMemReaderForUnsafeExternalSorter.java

```diff
@@ -35,7 +35,7 @@ public PMemReaderForUnsafeExternalSorter(
         this.numRecordsRemaining = numRecords - position/2;
         this.taskMetrics = taskMetrics;
         int readBufferSize = SparkEnv.get() == null? 8 * 1024 * 1024 :
-                (int) SparkEnv.get().conf().get(package$.MODULE$.MEMORY_SPILL_PMEM_READ_BUFFERSIZE());
+                (int) (long) SparkEnv.get().conf().get(package$.MODULE$.MEMORY_SPILL_PMEM_READ_BUFFERSIZE());
```

Comment:
Use ```getLong```? https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/SparkConf.scala#L421

Reply:
This code is referring other similar usages in Vanilla Spark.

Reply:
Interesting, didn't think up why they didn't use a getLong API? That seems better.